### PR TITLE
Drop support for .NET 3.5

### LINKED ISSUE #217: Drop support for .NET 3.5
I'm not sure whether this time has come yet but I thought I'd start the discussion and give the community an opportunity to comment.

.NET 4.0 has been around for almost 4 years now.

Removing the .NET 3.5 project will lessen our maintenance burden, reduce build times, package size, repo size, etc.

### PR DESCRIPTION
**Pull request by thomaslevesque**
I don't think I ever sent a PR with so much red in it :smile: 

Fixes #217

### CODE REVIEW

**Comment by blairconrad** [comment]
Hey, @thomaslevesque, while you're culling stuff, I hear there's a terrible approximation of `HasFlag` [over in FakeConfigurationManager](https://github.com/FakeItEasy/FakeItEasy/blob/master/src/FakeItEasy/Configuration/FakeConfigurationManager.cs#L88-L92). It probably should've been in a shim, but wasn't.

**Comment by blairconrad** [comment]
>

**Comment by blairconrad** [comment]
>  I don't think I ever sent a PR with so much red in it

Congratulations. My goal has always been to match @danielmarbach's [add/remove ratio](https://github.com/FakeItEasy/FakeItEasy/graphs/contributors).

File: src/FakeItEasy/ReturnValueConfigurationExtensions.cs [hunk 1]
```diff
@@ -342,6 +323,5 @@ public static void ReturnsNextFromSequence<T>(this IReturnValueConfiguration<Tas
 
             configuration.ReturnsLazily(x => queue.Dequeue()).NumberOfTimes(queue.Count);
         }
-#endif
     }
 }
```
Comment (blairconrad) [inline src/FakeItEasy/ReturnValueConfigurationExtensions.cs]: I don't even care if this file compiles anymore, I love the new version so much more.

**Comment by blairconrad** [comment]
Should we still define the `NET40` constant in FakeItEasy.csproj? If it's the done thing, that's fine, but we're not using it anymore. If we introduce additional compilation constants later, do we want them to be platform-based, or feature-based (e.g. SUPPORTS_SERIALIZATION)?

**Comment by danielmarbach** [comment]
:) wow! I'm surprised by myself ;)

**Comment by thomaslevesque** [comment]
> Hey, @thomaslevesque, while you're culling stuff, I hear there's a terrible approximation of HasFlag over in FakeConfigurationManager. It probably should've been in a shim, but wasn't.

Why "terrible"? I think it's fine. The method still makes sense, even if it's just for readability.

> Should we still define the NET40 constant in FakeItEasy.csproj?

I considered it, but couldn't see any drawback in keeping it. I can remove it if you prefer.

**Comment by adamralph** [comment]
> Should we still define the NET40 constant in FakeItEasy.csproj?

If there's no longer any `#if` in the code which uses it, and if it's not used in any other way, then YAGNI, we should remove it.

**Comment by adamralph** [comment]
LGTM.

@blairconrad any remaining concerns?

I'm happy to merge now. The `NET40` constant can be removed now or later, I don't mind when.

**Comment by thomaslevesque** [comment]
I removed the NET40 compilation symbol (amended the commit)

**Comment by adamralph** [comment]
We can still address https://github.com/FakeItEasy/FakeItEasy/pull/687#issuecomment-215602233 in another PR, if it's an issue.

**Comment by blairconrad** [comment]
> blairconrad any remaining concerns?

@adamralph, I do wake up early, but you asked at 5:29 and merged at 6:23 local. Give a guy a chance to finish [a bowl of cereal](https://www.allbran.ca/en_CA/products/buds-cereal.html?gclid=Cj0KEQjwjIy5BRClh8m_9Zu64d8BEiQAtZsQfzjxVfMZ1UPj8AAjPRMnwroEkcF-YRB2qCsI6_Po0DYaAuPZ8P8HAQ)!

Yeah, I had no concerns. My `HasFlag`ish comment is so minor that it's only worth addressing if someone's in the area and offended, IMO.

**Comment by adamralph** [comment]
@blairconrad yes, sorry, the 5:29 - 6:23 interval wasn't really my idea of an RFC period for you :stuck_out_tongue: I guessed that your concern is very minor and could be addressed separately so I went and ahead and merged without waiting for your reply. Hope that was OK.

**Comment by blairconrad** [comment]
Of course, @adamralph. I was just having a little fun with you. Sorry.

**Pull request merged**